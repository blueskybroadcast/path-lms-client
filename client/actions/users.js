import normalize from 'jsonapi-normalizer';

import { FETCH_USERS_SUCCESS } from './types';

export const fetchUsers = () => async (dispatch, getState, api) => {
  const res = await api.get('/users');

  dispatch({
    type: FETCH_USERS_SUCCESS,
    payload: normalize(res.data)
  });
};
