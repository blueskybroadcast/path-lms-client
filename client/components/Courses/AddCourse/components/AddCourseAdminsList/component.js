import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown } from 'semantic-ui-react';

import AddCourseFormSection from '../AddCourseFormSection';

const AddCourseAdminsList = ({
  usersData, usersForDropdown, adminIds, handleAdminRemove, handleAdminSelect
}) => (
  <AddCourseFormSection
    label="Course Administrators"
    tooltip="You can assign administrator(s) for the course. Course administrators can manage the course and all course items related to the course."
  >
    <div className="controls">
      <div className="admin-tools add-admins" data-role="add-admins">
        <div className="admin-actions-wrapper" data-role="admin-controls">
          <div className="select-existing-users">
            <Dropdown
              multiple
              inline
              icon={false}
              trigger={
                <a className="launch-select" href="#">
                  Add Course Administrator
                  <div className="arrow" />
                </a>
              }
            >
              <Dropdown.Menu>
                <Dropdown.Menu scrolling>
                  {usersForDropdown.map((id) => {
                    const user = usersData[id];
                    return (
                      <Dropdown.Item
                        key={user.id}
                        text={`${user.firstName} ${user.lastName} (${user.email})`}
                        onClick={() => handleAdminSelect(user.id)}
                      />
                    );
                  })}
                </Dropdown.Menu>
              </Dropdown.Menu>
            </Dropdown>
          </div>
        </div>
      </div>
      <div className="option-list course-admins">
        <div className="header">
          <span>Users</span>
        </div>
        <ul data-role="course-admin-list">
          { adminIds.map((id) => {
            const user = usersData[id];
            return (
              <li key={id}>
                <span className="full-name">
                  {`${user.firstName} ${user.lastName}`}
                </span>
                <span className="email">
                  (
                  {user.email}
                  )
                </span>
                <span
                  className="remove-course-admin"
                  onClick={() => handleAdminRemove(user.id)}
                >
                  <i className="icon icon-times" />
                </span>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  </AddCourseFormSection>
);

AddCourseAdminsList.propTypes = {
  usersData: PropTypes.objectOf(PropTypes.object).isRequired,
  usersForDropdown: PropTypes.arrayOf(PropTypes.string).isRequired,
  adminIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleAdminRemove: PropTypes.func.isRequired,
  handleAdminSelect: PropTypes.func.isRequired
};

export default AddCourseAdminsList;
