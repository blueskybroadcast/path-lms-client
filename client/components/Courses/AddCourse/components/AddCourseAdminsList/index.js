import AddCourseAdminsList from './component';

export default AddCourseAdminsList;
