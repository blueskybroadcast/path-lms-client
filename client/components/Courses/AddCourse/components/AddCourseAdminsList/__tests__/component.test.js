import React from 'react';
import { shallow } from 'enzyme';
import AddCourseAdminsList from '../component';

import { usersData } from '../../../../../../fixtures/users';

describe('AddCourseAdminsList', () => {
  test('should render AddCourseAdminsList with data correctly', () => {
    const wrapper = shallow(
      <AddCourseAdminsList
        usersData={usersData}
        usersForDropdown={['2', '3']}
        adminIds={['2']}
        handleAdminRemove={jest.fn()}
        handleAdminSelect={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render AddCourseAdminsList without data correctly', () => {
    const wrapper = shallow(
      <AddCourseAdminsList
        adminIds={[]}
        usersForDropdown={[]}
        usersData={{}}
        handleAdminRemove={jest.fn()}
        handleAdminSelect={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
