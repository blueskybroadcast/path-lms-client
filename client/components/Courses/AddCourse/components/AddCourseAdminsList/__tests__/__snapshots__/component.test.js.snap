// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseAdminsList should render AddCourseAdminsList with data correctly 1`] = `
<AddCourseFormSection
  isRequired={false}
  label="Course Administrators"
  tooltip="You can assign administrator(s) for the course. Course administrators can manage the course and all course items related to the course."
>
  <div
    className="controls"
  >
    <div
      className="admin-tools add-admins"
      data-role="add-admins"
    >
      <div
        className="admin-actions-wrapper"
        data-role="admin-controls"
      >
        <div
          className="select-existing-users"
        >
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            deburr={false}
            icon={false}
            inline={true}
            minCharacters={1}
            multiple={true}
            noResultsMessage="No results found."
            openOnFocus={true}
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            trigger={
              <a
                className="launch-select"
                href="#"
              >
                Add Course Administrator
                <div
                  className="arrow"
                />
              </a>
            }
            wrapSelection={true}
          >
            <DropdownMenu>
              <DropdownMenu
                scrolling={true}
              >
                <DropdownItem
                  key="2"
                  onClick={[Function]}
                  text="Ted Baker (test@test.com)"
                />
                <DropdownItem
                  key="3"
                  onClick={[Function]}
                  text="Tom Ford (test2@test.com)"
                />
              </DropdownMenu>
            </DropdownMenu>
          </Dropdown>
        </div>
      </div>
    </div>
    <div
      className="option-list course-admins"
    >
      <div
        className="header"
      >
        <span>
          Users
        </span>
      </div>
      <ul
        data-role="course-admin-list"
      >
        <li
          key="2"
        >
          <span
            className="full-name"
          >
            Ted Baker
          </span>
          <span
            className="email"
          >
            (
            test@test.com
            )
          </span>
          <span
            className="remove-course-admin"
            onClick={[Function]}
          >
            <i
              className="icon icon-times"
            />
          </span>
        </li>
      </ul>
    </div>
  </div>
</AddCourseFormSection>
`;

exports[`AddCourseAdminsList should render AddCourseAdminsList without data correctly 1`] = `
<AddCourseFormSection
  isRequired={false}
  label="Course Administrators"
  tooltip="You can assign administrator(s) for the course. Course administrators can manage the course and all course items related to the course."
>
  <div
    className="controls"
  >
    <div
      className="admin-tools add-admins"
      data-role="add-admins"
    >
      <div
        className="admin-actions-wrapper"
        data-role="admin-controls"
      >
        <div
          className="select-existing-users"
        >
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            deburr={false}
            icon={false}
            inline={true}
            minCharacters={1}
            multiple={true}
            noResultsMessage="No results found."
            openOnFocus={true}
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            trigger={
              <a
                className="launch-select"
                href="#"
              >
                Add Course Administrator
                <div
                  className="arrow"
                />
              </a>
            }
            wrapSelection={true}
          >
            <DropdownMenu>
              <DropdownMenu
                scrolling={true}
              />
            </DropdownMenu>
          </Dropdown>
        </div>
      </div>
    </div>
    <div
      className="option-list course-admins"
    >
      <div
        className="header"
      >
        <span>
          Users
        </span>
      </div>
      <ul
        data-role="course-admin-list"
      />
    </div>
  </div>
</AddCourseFormSection>
`;
